### ISSUE TITLE
GQL error text appearing twice

### ISSUE THREAD

**Issue by Vultraz** [keystonejs/keystone#1615]
See below:
![image](https://user-images.githubusercontent.com/3558659/64664165-fd85cd00-d499-11e9-878a-f821fa1f0df5.png)

**Comment by MadeByMike** [comment]
@Vultraz This is the intended error when someone tries to retrieve content from the AdminUI but is limited by access controls.

We want to limit situations where people see this, but it's not always unintended. To give you an example, you might have access to a list where an item has a relations to a list you don't have access to. Should you see that item, and error or list of related items appear empty?

We might not have got all this right so it might be helpful to know how you got to the error. But I'm going to close this issue as it's not generally a redundant error.

**Comment by Vultraz** [comment]
I only thought it was redundant since the text was repeated twice 😬 I got it while disabling the auth strategy so I could add an initial User entry but forgot to remove access controls as well.

**Comment by MadeByMike** [comment]
Hahaha! I didn't even notice. Sorry for miss-understanding :) I'll have a look into that!

**Comment by MadeByMike** [comment]
My guess is it is collating the errors and you happened to get this error twice for one request. Maybe we should make sure these are unique.

**Comment by singhArmani** [comment]
@MadeByMike : I was able to reproduce this issue as mentioned by @Vultraz. 
##Steps to reproduce
This can be reproduced in a standard starter app "Starter (Users + Authentication)" created with `yarn create keystone-app`
1. **Create a starter app** with **mongoose** adapter
<img width="580" alt="starter project" src="https://user-images.githubusercontent.com/11068601/73501209-30186e00-4419-11ea-91a6-e3325dce4c76.png">
2. Go to index.js file inside your newly created project directory and disable the authStrategy

```javascript
module.exports = {
  keystone,
  apps: [
    new GraphQLApp(),
    new AdminUIApp({
      enableDefaultRoute: true,
      // authStrategy,
    }),
  ],
};
```

3. Go to `http://localhost:3000/admin/users` and a duplicate **GraphQL** error should appear
![image](https://user-images.githubusercontent.com/11068601/73501959-88506f80-441b-11ea-85d6-346279f27ff7.png)

## Additional Context
The network tab in chrome shows the following response preview which seems to carry an array of two messages: 
![image](https://user-images.githubusercontent.com/11068601/73502190-45db6280-441c-11ea-811d-5eb34984d86a.png)

**Comment by singhArmani** [comment]
Following with the issue, the query we are generating from the client side (app-admin-ui) looks like: 
```js
{
  allUsers(first: 50, orderBy: "name_ASC") {
    id
    _label_
    name
    email
  }
  _allUsersMeta(search: "") {
    count
  }
}
```
Where the response from the GraphQL server is: 
```js
{
  "errors": [
    {
      "message": "You do not have access to this resource",
      "name": "AccessDeniedError",
      "time_thrown": "2020-02-03T00:43:42.041Z",
      "data": {
        "type": "query",
        "target": "_allUsersMeta"
      },
      "path": [
        "_allUsersMeta",
        "count"
      ],
      "uid": "ck65qdtui0005c40dfsrugqau"
    },
    {
      "message": "You do not have access to this resource",
      "name": "AccessDeniedError",
      "time_thrown": "2020-02-03T00:43:42.040Z",
      "data": {
        "type": "query",
        "target": "allUsers"
      },
      "path": [
        "allUsers"
      ],
      "uid": "ck65qdtum0006c40d02b80ua4"
    }
  ],
  "data": {
    "allUsers": null,
    "_allUsersMeta": {
      "count": null
    }
  }
}
```

As we can see we are also querying for `_allUserMeta` field, and thus getting two error messages. Additionally, the source code of generating the **ApolloError** instance is looping over each error message and concatenating them together to form a single message. Here is the [line](https://github.com/apollographql/apollo-client/blob/5b6e751636f23d9798172dd81c68645e82a93ce8/src/errors/ApolloError.ts#L17).  

Here's is the response that comes back when use **useQuery** in **admin-ui** (client/pages/List). 
![image](https://user-images.githubusercontent.com/11068601/73618664-44f53b80-467d-11ea-918c-00a1aa057ed0.png)

For rendering the error for the list page, we are consuming the message as: 
```js
 message = error.message;
```

To summarise this, we are naively using the message from the response. Ideally, IMO we should be using a better error mechanism to deal with these situation. Open for suggestions.
